### PR TITLE
Adds a loss function into the fitting module to aid in validation of the model

### PR DESCRIPTION
**Pull request by zifn**
Addresses #19 by implementing a cross-entropy loss function using the Y_lm probability density models as a classifier to aid in model evaluation.

### CODE REVIEW

File: src/fitting.py [hunk 1]
```diff
@@ -167,3 +167,49 @@ def fit_Y_lms_binning_least_squares(M_xyz, L_max, numb_bins, only_even_Ls=False)
...
+    """
+    computes the cross entropy using the shpericial harmonic distribution and labeled data
+    from clustering
+
+    Parameters
+    ------------
+    data_val_xyz: array Nx3
+        Validation data used to compute the cross entropy of the clasification
+    labels: array of ints
+        Each entry varries from 0 to (number classes - 1). Used to reference the model params.
```
Comment (sajantanand) [inline src/fitting.py:181]: What happens to data points with '-1' label? This is needed for optics clustering.

File: src/fitting.py [hunk 2]
```diff
@@ -167,3 +167,49 @@ def fit_Y_lms_binning_least_squares(M_xyz, L_max, numb_bins, only_even_Ls=False)
...
+        maximum L value used to generate a truncated array of Y_lms
+    only_even_Ls: bool
+        if in the molecular frame, only use even L values to generate features
+
+    Returns
+    --------
+    cross_entropy: float
+        the cross entropy given labeled validation data
+    """
+    assert data_val_xyz.shape[1] == 3
```
Comment (sajantanand) [inline src/fitting.py:194]: Is this not going to be 15?

Comment (sajantanand) [reply src/fitting.py:194]: Perhaps I don't understand what this array is supposed to represent.

File: src/tests/test_fitting.py [hunk 3]
```diff
@@ -84,3 +84,33 @@ def test_fit_Y_lms_va_binning():
...
+
+    samples = np.array([np.ones(N_valid),
+                                    rng.choice(theta, size=N_valid, replace=True),
+                                    rng.choice(phi, size=N_valid, replace=True)]).T
+    samples = fitting.spherical_to_cart(samples)
+    labels = np.zeros(N_valid)
+    model_params = [sample_B_lms]
+
+    entropy = fitting.validation_cross_entropy(samples, labels, model_params, L_max, only_even_Ls=False)
+    assert np.isclose(entropy, 0)
```
Comment (sajantanand) [inline src/tests/test_fitting.py:116]: Why do we expect a cross entropy of 0?

**Review by sajantanand — COMMENTED** [review]
I left some comments of things I don't understand. I don't think there is anything to correct. Just I am not sure why certain things are done the way they are.

File: src/tests/test_fitting.py (hunk 3 above)
Comment (zifn) [reply src/tests/test_fitting.py:116]: I'm only feeding in one class and one set of B_lms so the q probability of the point being in the class is 1 leading to an entropy of 0

**Review by sajantanand — APPROVED** [review]
Looks good to me.
